source/frontend/src/features/mapSideBar/compensation/detail/CompensationRequisitionHistoricalDetailContainer.tsx

```diff
@@ -0,0 +1,350 @@
+import moment from 'moment';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { useAcquisitionProvider } from '@/hooks/repositories/useAcquisitionProvider';
+import { useFinancialCodeRepository } from '@/hooks/repositories/useFinancialCodeRepository';
+import { useLeaseRepository } from '@/hooks/repositories/useLeaseRepository';
+import { useProductProvider } from '@/hooks/repositories/useProductProvider';
+import { useProjectProvider } from '@/hooks/repositories/useProjectProvider';
+import { useCompensationRequisitionRepository } from '@/hooks/repositories/useRequisitionCompensationRepository';
+import { ApiGen_CodeTypes_FileTypes } from '@/models/api/generated/ApiGen_CodeTypes_FileTypes';
+import { ApiGen_Concepts_AcquisitionFile } from '@/models/api/generated/ApiGen_Concepts_AcquisitionFile';
+import { ApiGen_Concepts_CompensationFinancial } from '@/models/api/generated/ApiGen_Concepts_CompensationFinancial';
+import { ApiGen_Concepts_CompensationRequisition } from '@/models/api/generated/ApiGen_Concepts_CompensationRequisition';
+import { ApiGen_Concepts_CompReqAcqPayee } from '@/models/api/generated/ApiGen_Concepts_CompReqAcqPayee';
+import { ApiGen_Concepts_CompReqLeasePayee } from '@/models/api/generated/ApiGen_Concepts_CompReqLeasePayee';
+import { ApiGen_Concepts_FileProperty } from '@/models/api/generated/ApiGen_Concepts_FileProperty';
+import { ApiGen_Concepts_FinancialCode } from '@/models/api/generated/ApiGen_Concepts_FinancialCode';
+import { ApiGen_Concepts_Lease } from '@/models/api/generated/ApiGen_Concepts_Lease';
+import { ApiGen_Concepts_Product } from '@/models/api/generated/ApiGen_Concepts_Product';
+import { ApiGen_Concepts_Project } from '@/models/api/generated/ApiGen_Concepts_Project';
+import { isValidId } from '@/utils';
+
+import { useGenerateH120 } from '../../acquisition/common/GenerateForm/hooks/useGenerateH120';
+import { CompensationRequisitionDetailViewProps } from './CompensationRequisitionDetailView';
+
+export interface ICompensationRequisitionHistoricalDetailContainerProps {
+  fileType: ApiGen_CodeTypes_FileTypes;
+  parentFileId: number;
+  compensationRequisitionId: number;
+  time: string;
+  clientConstant: string;
+  loading: boolean;
+  setEditMode: (editMode: boolean) => void;
+  View: React.FunctionComponent<CompensationRequisitionDetailViewProps>;
+}
+
+export const CompensationRequisitionHistoricalDetailContainer: React.FunctionComponent<
+  React.PropsWithChildren<ICompensationRequisitionHistoricalDetailContainerProps>
+> = ({
+  fileType,
+  parentFileId,
+  compensationRequisitionId,
+  time: almostTime,
+  clientConstant,
```

Comment:
does this need to be historic as well?

Reply:
I thought the constants where not going to change, however It would be safer to return the historic versions too.